typing/typecore.ml

```diff
@@ -3693,6 +3698,204 @@ and type_expect_
             exp_env = env;
             exp_attributes = sexp.pexp_attributes; }
 
+  | Pexp_functor (name, pack_opt, sbody) ->
```

Comment:
I think it'd be nice to move all this to a dedicated function, as is done with `Pexp_fun`, `Pexp_function` and `type_function` (relatedly, I think it'd make a bit more sense to move this case just after those, instead of having it at the end).

Reply:
Good idea, will do.